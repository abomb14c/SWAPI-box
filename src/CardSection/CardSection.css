.card-section{
    padding-right: 2%;
    margin-bottom:10%;
    width: 100%;
}