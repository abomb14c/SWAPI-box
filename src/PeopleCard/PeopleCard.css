*,
*:after,
*:before {
box-sizing: border-box;
margin: 0;
}

hr {
    margin:0 auto;
    text-align: center;
    width: 60%;
}

.person-card {
    background-color: #ffffff;
    border: 2px solid #D6D6D6;
    display: inline-block;
    height: 400px;
    left: 15px;
    margin: 3px;
    position: relative;
    width: 340px;
}

.person-name {
    color: rgba(255,255,255);
    margin: none;
    text-align: center;
    text-shadow: 4px 4px #000;
}

.name-section {
    background-image: url('../assets/peopler-header-section.jpg');
    background-size: cover;
    font-size: 36px;
    height: 125px;
    padding-top: 50px;
}

.content-div{
    margin: 15px;
}

.description-title{
    color: #EB544D;
    margin-bottom: 1%;
}

.description-content {
    font-size: 24px;
}