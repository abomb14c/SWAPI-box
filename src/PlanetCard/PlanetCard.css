*,
*:after,
*:before {
box-sizing: border-box;
margin: 0;
}

hr {
    text-align: center;
    margin:0 auto;
    width: 60%;
}

.planet-card {
background-color: #ffffff;
border: 2px solid #D6D6D6;
display: inline-block;
height: 500px;
left: 15px;
margin: 3px;
position: relative;
vertical-align: top;
width: 340px;
}

.planet-name {
    text-align: center;
    color: rgba(255,255,255);
    margin: none;
    text-shadow: 4px 4px #000;
}

.planet-name-section {
    height: 125px;
    background-image: url('../assets/planets-section-header.jpg');
    background-size: cover;
    padding-top: 50px;
    font-size: 36px;
}

.content-div{
    margin: 15px;
}

.description-title{
    color: #EB544D;
    margin-bottom: 1%;
}

.description-content {
    font-size: 24px;
}

.residents{
    font-size: 18px;
}