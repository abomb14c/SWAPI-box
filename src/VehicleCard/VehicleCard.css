*,
*:after,
*:before {
box-sizing: border-box;
margin: 0;
}

hr {
    text-align: center;
    margin:0 auto;
    width: 60%;
}

.description-title{
    color: #EB544D;
    margin-bottom: 1%;
}

.description-content {
    font-size: 24px;
}

.content-div {
    margin: 15px;
}

.vehicle-card {
    background-color: #ffffff;
    border: 2px solid #D6D6D6;
    display: inline-block;
    height: 400px;
    left: 15px;
    margin: 3px;
    position: relative;
    vertical-align: top;
    width: 340px;
}

.vehicle-name {
    color: rgba(255,255,255);
    margin: none;
    text-align: center;
    text-shadow: 4px 4px #000;
}

.vehicle-name-section {
    background-image: url('../assets/vehicles-section-header.jpg');
    background-size: cover;
    font-size: 36px;
    height: 140px;
    padding-top: 50px;
}



.residents{
    font-size: 18px;
}
