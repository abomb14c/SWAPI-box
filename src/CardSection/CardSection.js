import React from 'react'
import PeopleCard from '../Card/PeopleCard'


const CardSection = ({people}) => {
       const displayPeople= people.map(person => {
           return <PeopleCard {...person} />
       })
   
    
    return(
        <section className="card-section">
            {displayPeople}
        </section>
    )
}

export default CardSection