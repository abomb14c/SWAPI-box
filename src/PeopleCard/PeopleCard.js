import React from 'react';
import './PeopleCard.css';
import propTypes from 'prop-types';

const PeopleCard = (props) => {
  const{name, planet, species, population, findFavoritePerson, id, category} = props;
 
  return (
    <article className="person-card">
      <section className="name-section">
        <h1 className="person-name">{name}</h1>
        <button onClick={()=>findFavoritePerson(id)} className="favorites">Favorite</button>
      </section>
      <div className="content-div">
        <p className="description-title">Origin</p>
        <h3 className="description-content">{planet}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Species</p>
        <h3 className="description-content">{species}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Population</p>
        <h3 className="description-content">{population}</h3>
      </div>
    </article>
  );
};

PeopleCard.propTypes = {
  name: propTypes.string,
  planet: propTypes.string,
  species: propTypes.string,
  population: propTypes.number,
  findFavoritePerson: propTypes.func,
  id: propTypes.string,
  category: propTypes.string
};
export default PeopleCard;