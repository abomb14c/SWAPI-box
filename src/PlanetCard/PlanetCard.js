import React from 'react';
import './PlanetCard.css';
import PropTypes from 'prop-types';

const PlanetCard = (props) => {
  const{name, terrain, population, climate, residents, findFavoritePlanet, id, category} = props;

  return (
    <article className="planet-card">
      <section className="planet-name-section">
        <h1 className="planet-name">{name}</h1>
        <button onClick={()=> findFavoritePlanet(id)} className="favorites">Favorite</button>
      </section>
      <div className="content-div">
        <p className="description-title">Terrain</p>
        <h3 className="description-content">{terrain}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Population</p>
        <h3 className="description-content">{population}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Climate</p>
        <h3 className="description-content">{climate}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Residents</p>
        <h3 className="description-content residents">{residents}</h3>
      </div>
    </article>
  );
     
};

PlanetCard.propTypes = {
  name: PropTypes.string,
  terrain: PropTypes.string,
  population: PropTypes.number,
  climate: PropTypes.string,
  residents: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array
  ]),
  findFavoritePlanet: PropTypes.func,
  id: PropTypes.string,
  category: PropTypes.string
};
 
export default PlanetCard;
