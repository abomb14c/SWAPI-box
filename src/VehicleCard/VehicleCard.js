import React from 'react';
import './VehicleCard.css';
import PropTypes from 'prop-types';

const VehicleCard = (props) => {
  const{name, model, type, passengers, findFavoriteVehicle, id, category} = props;

  return (
    <article className="vehicle-card">
      <section className="vehicle-name-section">
        <h1 className="vehicle-name">{name}</h1>
        <button onClick={()=>findFavoriteVehicle(id)} className="favorites">Favorite</button>
      </section>
      <div className="content-div">
        <p className="description-title">Model</p>
        <h3 className="description-content">{model}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Class</p>
        <h3 className="description-content">{type}</h3>
      </div>
      <hr></hr>
      <div className="content-div">
        <p className="description-title">Passengers</p>
        <h3 className="description-content">{passengers}</h3>
      </div>
    </article>
  );
    
};
VehicleCard.propTypes = {
  name: PropTypes.string,
  model: PropTypes.string,
  type: PropTypes.string,
  passengers: PropTypes.number,
  findFavoriteVehicle: PropTypes.func,
  id: PropTypes.string,
  category: PropTypes.string
};

export default VehicleCard;



