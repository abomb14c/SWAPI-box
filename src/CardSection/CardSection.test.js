import React from 'react';
import ReactDOM from 'react-dom';
import CardSection from './CardSection';
import PeopleCard from '../PeopleCard/PeopleCard';
import PlanetCard from '../PlanetCard/PlanetCard';
import VehicleCard from '../VehicleCard/VehicleCard';
import {shallow, mount} from 'enzyme';

describe('App', () => {
  let wrapper;
  let mockPeople = [{name:"Alan", planet:"Earth"}];
  let mockActiveCategory= '';
  let mockPlanets= [{name:"earth", climate: "slowly dying"}];
  let mockVehicles=[{name:'subaru', passangers:2}];
  let mockFavorite= jest.fn();
  let mockFavoritesArray = [{name:"Alan", planet:"Earth"}, 
    {name:"earth", climate: "slowly dying"}, 
    {name:'subaru', passangers:2}];

  beforeEach(() => {
    wrapper = shallow(<CardSection people={mockPeople}
      activeCategory={mockActiveCategory}
      planets={mockPlanets}
      vehicles={mockVehicles}
      findFavoritePerson={mockFavorite}
      findFavoriteVehicle={mockFavorite}
      findFavoritePlanet={mockFavorite}
      favoritesArray={mockFavoritesArray}
    />);
  });
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the people cards', () => {
    wrapper= mount(<CardSection people={mockPeople} activeCategory={'is people'} />);
        
    const peopleCard= wrapper.find("section").find("article");
    expect(peopleCard.hasClass("person-card")).toBe(true);
  });

  it('should render the planet cards', () => {
    wrapper= mount(<CardSection planets={mockPlanets} activeCategory={'is planets'} />);
        
    const planetCard= wrapper.find("section").find("article");
    expect(planetCard.hasClass("planet-card")).toBe(true);
  });

  it('should render the vehicle cards', () => {
    wrapper= mount(<CardSection vehicles={mockVehicles} activeCategory={'is vehicles'} />);
        
    const vehicleCard= wrapper.find("section").find("article");
    expect(vehicleCard.hasClass("vehicle-card")).toBe(true);
  });
});