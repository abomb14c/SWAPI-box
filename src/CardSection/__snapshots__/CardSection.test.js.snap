// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardSection
    activeCategory=""
    favoritesArray={
        Array [
            Object {
              "name": "Alan",
              "planet": "Earth",
            },
            Object {
              "climate": "slowly dying",
              "name": "earth",
            },
            Object {
              "name": "subaru",
              "passangers": 2,
            },
          ]
    }
    findFavoritePerson={[Function]}
    findFavoritePlanet={[Function]}
    findFavoriteVehicle={[Function]}
    people={
        Array [
            Object {
              "name": "Alan",
              "planet": "Earth",
            },
          ]
    }
    planets={
        Array [
            Object {
              "climate": "slowly dying",
              "name": "earth",
            },
          ]
    }
    vehicles={
        Array [
            Object {
              "name": "subaru",
              "passangers": 2,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": undefined,
      "className": "card-section",
    },
    "ref": null,
    "rendered": null,
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": "card-section",
      },
      "ref": null,
      "rendered": null,
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
